recirq/time_crystals/dtctask.py

```diff
@@ -0,0 +1,254 @@
+        """ Yield a sequence of DTCTasks that are the product of the options in self.options_dict.
+                All DTCTask attributes not in options_dict are taken to be their default values
+        Yields:
+            DTCTasks with parameters taken from self.options_dict
+        """
+
+        # take product over elements of options_dict, in the order of options_order
+        for components in itertools.product(*(self.options_dict[attribute_name] for attribute_name in self.options_order)):
+            # prepare arguments for DTCTask
+            kwargs = dict(zip(self.options_order, components))
+            yield DTCTask(qubits=self.qubits, disorder_instances=self.disorder_instances, **kwargs)
+
+
+@recirq.json_serializable_dataclass(namespace='recirq.readout_scan',
+                                    registry=recirq.Registry,
+                                    frozen=False)
+class DTCTask:
+    """ A task for managing inputs to a Discrete Time Crystal experiment, over some number of disorder instances
+
+    Attributes:
+        dataset_id: unique identifier for this dataset
+        qubits: a chain of connected qubits available for the circuit
+        disorder_instances: number of disorder instances averaged over
+        initial_states: initial state of the system used in circuit
+        g: thermalization constant used in circuit
+        local_fields: random noise used in circuit
+        thetas: theta parameters for FSim Gate used in circuit
+        zetas: zeta parameters for FSim Gate used in circuit
+        chis: chi parameters for FSim Gate used in circuit
+        phis: phi parameters for FSim Gate used in circuit
+        gammas: gamma parameters for FSim Gate used in circuit
+
+    """
+    # Task parameters
+    dataset_id: str
+
+    # experiment parameters
+    qubits: Sequence[cirq.Qid]
+    disorder_instances: int
+
+    # FSim Gate parameters
+    # ndarrays in this section are in shape (disorder_instances, len(qubits) - 1)
+    g: int
+    initial_states: np.ndarray
+    local_fields: np.ndarray
+
+    # FSim Gate Parameters
+    # ndarrays in this section are in shape (disorder_instances, len(qubits) - 1)
+    thetas: np.ndarray
+    zetas: np.ndarray
+    chis: np.ndarray
+    gammas: np.ndarray
+    phis: np.ndarray
+
+
+    def __init__(
+            self,
+            qubits: Optional[Sequence[cirq.Qid]] = None,
+            disorder_instances: Optional[int] = None,
+            g: Optional[int] = None,
+            initial_state: Optional[np.ndarray] = None,
+            initial_states: Optional[np.ndarray] = None,
+            local_fields: Optional[np.ndarray] = None,
+            thetas: Optional[np.ndarray] = None,
+            zetas: Optional[np.ndarray] = None,
+            chis: Optional[np.ndarray] = None,
+            gammas: Optional[np.ndarray] = None,
+            phis: Optional[np.ndarray] = None
+            ):
+
+        self.dataset_id = datetime.datetime.utcnow()
```

Comment:
avoid doing business logic in the constructor. dataclasses ideally shouldn't define `__init__` at all